granulate_utils/metadata/databricks_client.py

```diff
@@ -56,53 +59,49 @@ def get_webui_address() -> Optional[str]:
             raise DatabricksJobNameDiscoverException(f"Failed to get Databricks webui address {properties=}") from e
         return f"{host}:{DEFAULT_WEBUI_PORT}"
 
-    def get_job_name(self) -> Optional[str]:
+    def extract_relevant_metadata(self) -> Optional[Dict[str, str]]:
         # Retry in case of a connection error, as the metrics server might not be up yet.
         start_time = time.monotonic()
         while time.monotonic() - start_time < DATABRICKS_JOBNAME_TIMEOUT_S:
             try:
-                if cluster_metadata := self._cluster_all_tags_metadata():
-                    name = self._get_name_from_metadata(cluster_metadata)
-                    if name:
-                        self.logger.debug("Found name in metadata", job_name=name, cluster_metadata=cluster_metadata)
-                        return name
-                    else:
-                        self.logger.debug("Failed to extract name from metadata", cluster_metadata=cluster_metadata)
-                        return None
+                if cluster_all_props := self._cluster_all_tags_metadata():
+                    self.logger.info(
+                        "Successfully got relevant cluster tags metadata", cluster_all_props=cluster_all_props
+                    )
+                    return cluster_all_props
                 else:
-                    # No job name yet, retry.
+                    # No environment metadata yet, retry.
                     time.sleep(RETRY_INTERVAL_S)
             except DatabricksJobNameDiscoverException:
-                self.logger.exception("Failed to get Databricks job name")
+                self.logger.exception("Failed to get DBX environment properties")
                 return None
             except Exception:
-                self.logger.exception("Generic exception was raise during spark job name discovery")
+                self.logger.exception("Generic exception was raise during DBX environment properties discovery")
                 return None
         self.logger.info("Databricks get job name timeout, continuing...")
         return None
 
-    @staticmethod
-    def _get_name_from_metadata(metadata: Dict[str, str]) -> Optional[str]:
-        if JOB_NAME_KEY in metadata:
-            return str(metadata[JOB_NAME_KEY]).replace(" ", "-").lower()
-        return None
-
-    def _cluster_all_tags_metadata(self) -> Optional[Dict[str, str]]:
+    def _discover_apps_url(self) -> bool:
```

Comment:
What about changing this function to be `_get_apps_url` and it will return an `Optional[str]` ? Then we can just call it from `_cluster_all_tags_metadata` and check if we were able to get it?

Reply:
We're discovering an object property
we should discover it, and validate it was discovered.
I don't think we should call `_get_apps_url` where we do know that the property might not have a value yet.
please LMK if you still think differently